src/platform/plugins/shared/controls/public/control_group/selections_manager.ts

```diff
@@ -22,8 +22,8 @@ import { ControlGroupApi } from './types';
 export function initSelectionsManager(
   controlGroupApi: Pick<ControlGroupApi, 'autoApplySelections$' | 'children$' | 'untilInitialized'>
 ) {
-  const filters$ = new BehaviorSubject<Filter[] | undefined>([]);
-  const unpublishedFilters$ = new BehaviorSubject<Filter[] | undefined>([]);
+  const filters$ = new BehaviorSubject<Filter[] | undefined>(undefined);
```

Comment:
Can you provide some context on the cause of the problem and how switching from `[]` to `undefined` resolved the problem?

Reply:
On our side, we start with having filters as undefined and then set them according to URL state/local storage. Having the empty array here makes it have 3 values for the filters each time:

1. undefined
2. [] -> which is the default value of `filters$`
3. [{}] -> actual filter saved in the local storage

I am removing the default value of `[]` in order to ensure we start with undefined and only have this value changed when the consumer passes a value.

Reply:
This is causing a problem when "auto apply" is disabled. To view the problem
1) open web logs sample data dashboard
2) controls -> settings and uncheck "Apply selections automatically"
3) save dashboard.
4) open dashboard. Notice how the apply button is enabled because controls think they have unapplie
![Uploading Screenshot 2025-04-03 at 8.47.53 AM.png…]()
d selections

Reply:
Thanks for catching this issue!

Fixed in [de7839b](https://github.com/elastic/kibana/pull/214612/commits/de7839bce1b80572929faabbf1226de4f01c71e0)